geotrek/api/tests/test_v2.py

```diff
@@ -1839,3 +1839,73 @@ def test_web_links_in_trek_detail(self):
         self.assertEqual(response.json()['web_links'][0]['category']['label']['en'], "Category")
         self.assertEqual(response.json()['web_links'][0]['category']['id'], self.web_link_cat.pk)
         self.assertEqual(response.json()['web_links'][0]['category']['pictogram'], 'http://testserver/media/dummy_picto.png')
+
+
+class TrekDifficultyFilterCase(TestCase):
+    @classmethod
+    def setUpTestData(cls):
+        cls.v_easy = trek_factory.DifficultyLevelFactory(difficulty="Very easy")
+        cls.easy = trek_factory.DifficultyLevelFactory(difficulty="Easy")
+        cls.medium = trek_factory.DifficultyLevelFactory(difficulty="Medium")
+        cls.hard = trek_factory.DifficultyLevelFactory(difficulty="Very hard")
+        cls.v_hard = trek_factory.DifficultyLevelFactory(difficulty="Hard")
+        cls.trek_v_easy = trek_factory.TrekFactory(difficulty=cls.v_easy)
+        cls.trek_easy = trek_factory.TrekFactory(difficulty=cls.easy)
+        cls.trek_medium = trek_factory.TrekFactory(difficulty=cls.medium)
+        cls.trek_hard = trek_factory.TrekFactory(difficulty=cls.hard)
+        cls.trek_v_hard = trek_factory.TrekFactory(difficulty=cls.v_hard)
+
+    def assert_trek_is_in_reponse(self, response, expected_trek):
+        found = list(filter(lambda trek: trek['id'] == expected_trek.pk, response.json()['results']))
+        self.assertTrue(found)
+
+    def assert_trek_is_not_in_reponse(self, response, expected_trek):
+        found = list(filter(lambda trek: trek['id'] == expected_trek.pk, response.json()['results']))
+        self.assertFalse(found)
+
+    def test_difficulty_ids(self):
+        self.assertEqual(self.v_easy.id, 1)
+        self.assertEqual(self.easy.id, 2)
+        self.assertEqual(self.medium.id, 3)
+        self.assertEqual(self.hard.id, 4)
+        self.assertEqual(self.v_hard.id, 5)
+
+    def test_filter_difficulty_min(self):
+        response = self.client.get(f"/api/v2/trek/?difficulty_min={self.medium.id}")
```

Comment:
lors de l'utilisation du client de test de django, c'est plus propre de gérer les parametres GET et POST avec un dictionnaire
```suggestion
        response = self.client.get(f"/api/v2/trek/", {"difficulty_min": self.medium.id})
```

Reply:
idem partout